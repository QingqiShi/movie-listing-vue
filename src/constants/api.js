const urls = {
  API_KEY: "e8e8b3d77e2a85c1ae77d3c6a5fafc6e",
  API_HOST: "https://api.themoviedb.org/3",
  ORIGINAL_IMAGE_WIDTH: 2000,

  CONFIGURATION: () => `${urls.API_HOST}/configuration?api_key=${urls.API_KEY}`,

  /** @type {(language:string) => string} */
  NOW_PLAYING: language =>
    `${urls.API_HOST}/movie/now_playing?api_key=${
      urls.API_KEY
    }&language=${language}`,

  /** @type {(language:string) => string} */
  GENRES_LIST: language =>
    `${urls.API_HOST}/genre/movie/list?api_key=${
      urls.API_KEY
    }&language=${language}`,

  /** @type {(baseUrl:string, size:string, src:string) => string} */
  IMAGE: (baseUrl, size, src) => `${baseUrl}${size}${src}`
};

export default urls;
