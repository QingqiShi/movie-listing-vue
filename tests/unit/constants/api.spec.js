import api from "@/constants/api";

describe("api constants", () => {
  it("CONFIGURATION", () => {
    expect(api.CONFIGURATION()).toMatchSnapshot();
  });

  it("NOW_PLAYING", () => {
    expect(api.NOW_PLAYING("en_GB")).toMatchSnapshot();
  });

  it("GENRES_LIST", () => {
    expect(api.GENRES_LIST("en_GB")).toMatchSnapshot();
  });

  it("IMAGE", () => {
    expect(api.IMAGE("https:test.com/", "w100", "/test.jpg")).toMatchSnapshot();
  });
});
