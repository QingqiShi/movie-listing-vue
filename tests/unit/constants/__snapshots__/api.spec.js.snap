// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api constants CONFIGURATION 1`] = `"https://api.themoviedb.org/3/configuration?api_key=e8e8b3d77e2a85c1ae77d3c6a5fafc6e"`;

exports[`api constants GENRES_LIST 1`] = `"https://api.themoviedb.org/3/genre/movie/list?api_key=e8e8b3d77e2a85c1ae77d3c6a5fafc6e&language=en_GB"`;

exports[`api constants IMAGE 1`] = `"https:test.com/w100/test.jpg"`;

exports[`api constants NOW_PLAYING 1`] = `"https://api.themoviedb.org/3/movie/now_playing?api_key=e8e8b3d77e2a85c1ae77d3c6a5fafc6e&language=en_GB"`;
